alerter.py

```diff
@@ -1,25 +1,26 @@
-alert_failure_count = 0
+from alert import alert_in_celcius, call_network_alert
+threshold = 200
 
 def network_alert_stub(celcius):
     print(f'ALERT: Temperature is {celcius} celcius')
     # Return 200 for ok
     # Return 500 for not-ok
     # stub always succeeds and returns 200
-    return 200
+    status_code = 200
+    # Return 500 if exceeds threshold. Assuming threshold of 200.
+    if celcius > threshold:
+        status_code = 500
+    return status_code
 
-def alert_in_celcius(farenheit):
-    celcius = (farenheit - 32) * 5 / 9
-    returnCode = network_alert_stub(celcius)
-    if returnCode != 200:
-        # non-ok response is not an error! Issues happen in life!
-        # let us keep a count of failures to report
-        # However, this code doesn't count failures!
-        # Add a test below to catch this bug. Alter the stub above, if needed.
-        global alert_failure_count
-        alert_failure_count += 0
+# separated getting alert in celcius and network calling part 
+# passing network communication as an argument make it more easy for prod code.
+celcius = alert_in_celcius(400.5)
+alert_failure_count = call_network_alert(celcius, network_alert_stub)
+celcius = alert_in_celcius(303.6)
+alert_failure_count = call_network_alert(celcius, network_alert_stub)
 
+# test to check alert_failure_count
+assert(alert_failure_count > 0)

```

Comment:
this is weak test case. The test code should know exactly what is the failure count. Just >0 will not guarantee complete validation. what if the value is 1 instead of 2?
Also the test values should cover values at minimum below threshold, threshold and above threshold.

Reply:
@prithamvshetty I have made feedback changes. Also, I have separated prod code from test and stub. Please check if that is the right way.

Reply:
Checked and this is fine👍